src/components/z-book-card/index.tsx

```diff
@@ -0,0 +1,280 @@
+  hasResources = false;
+
+  @State()
+  showResources = false;
+
+  private id: string;
+
+  private moveFocusToResources = false;
+
+  private resourcesWrapper: HTMLDivElement;
+
+  @Listen("resize", {target: "window"})
+  @Listen("orientationchange", {target: "window"})
+  handleResize(): void {
+    this.isMobile = window.innerWidth <= mobileBreakpoint;
+  }
+
+  componentWillLoad(): void {
+    this.id = `id-${randomId()}`;
+    this.handleResize();
+  }
+
+  componentWillRender(): void {
+    this.handleResources();
+  }
+
+  componentDidRender(): void {
+    if (this.moveFocusToResources) {
+      this.resourcesWrapper.focus();
+      this.moveFocusToResources = false;
+    }
+  }
+
+  private handleResources(): void {
+    if (this.variant !== BookCardVariant.EXPANDED || !this.isMobile) {
+      return;
+    }
+    this.hasResources = this.hostElement.querySelectorAll("[slot=resources]")?.length > 0;
+  }
+
+  private toggleResources(): void {
+    this.showResources = !this.showResources;
+    if (this.showResources) {
+      this.moveFocusToResources = true;
+    }
+  }
+
+  private renderCard(): JSX.Element {
+    switch (this.variant) {
+      case BookCardVariant.EXPANDED:
+        return this.isMobile ? this.renderMobileExpandedCard() : this.renderExpandedCard();
+      case BookCardVariant.COMPACT:
+        return <div />;
+      case BookCardVariant.SEARCH:
+        return <div />;
+    }
+  }
+
+  private renderExpandedCard(): JSX.Element {
+    return (
+      <div class="wrapper">
+        {this.renderCover()}
+        <div class="content">
+          <div class="top">
+            <div class="info">
+              <div class="left">
+                {this.renderAuthors()}
+                {this.renderOperaTitle()}
+                {this.renderVolumeTitle()}
+                {this.renderIsbn()}
+              </div>
+              <div class="right">{this.renderHeaderCtaSlot()}</div>
+            </div>
+            {this.renderTagsSlot()}
+          </div>
+          <div class="bottom">{this.renderResourcesSlot()}</div>
+        </div>
+      </div>
+    );
+  }
+
+  private renderMobileExpandedCard(): JSX.Element {
+    return (
+      <div class="wrapper">
+        <div class="header">
+          {this.renderOperaTitle()}
+          {this.renderHeaderCtaSlot()}
+        </div>
+        {this.renderCover()}
+        <div class="content">
+          {this.renderTagsSlot()}
+          {this.renderAuthors()}
+          {this.renderVolumeTitle()}
+          {this.renderIsbn()}
+        </div>
+        {this.hasResources && (
+          <div class={`footer ${this.showResources ? "open" : "close"}`}>
+            {this.renderResourcesSlot()}
+            {this.renderShowResources()}
+          </div>
+        )}
+      </div>
+    );
+  }
+
+  private renderCover(): JSX.Element {
+    return (
+      <div class="cover">
+        {this.ribbon && this.variant !== BookCardVariant.COMPACT && <div class="ribbon">{this.ribbon}</div>}
```

Comment:
Se mettiamo il testo di `this.ribbon` dentro uno span, poi usando `flex` e `align-items` sul `div.ribbon` si dovrebbe riuscire a centrare meglio verticalmente il testo.